typescript/sdk/src/providers/transactions/transformer/ethersV5/EV5InterchainAccountTxTransformer.ts

```diff
@@ -21,35 +26,48 @@ export class EV5InterchainAccountTxTransformer
   constructor(
     public readonly multiProvider: MultiProvider,
     public readonly props: EV5InterchainAccountTxTransformerProps,
-  ) {}
+  ) {
+    assert(
+      this.props.config.localRouter,
+      'Invalid AccountConfig: Cannot retrieve InterchainAccount.',
+    );
+  }
 
   public async transform(
     ...txs: PopulatedTransaction[]
-  ): Promise<PopulatedTransaction[]> {
-    const txChainsToInnerCalls: Record<ChainName, CallData[]> = {};
-
-    txs.map(({ to, data, value, chainId }: PopulatedTransaction) => {
-      assert(to, 'Invalid PopulatedTransaction: Missing to field');
-      assert(data, 'Invalid PopulatedTransaction: Missing data field');
-      assert(chainId, 'Invalid PopulatedTransaction: Missing chainId field');
-      const txChain = this.multiProvider.getChainName(chainId);
-      if (!txChainsToInnerCalls[txChain]) txChainsToInnerCalls[txChain] = [];
-      txChainsToInnerCalls[txChain].push({ to, data, value });
-    });
-
-    const transformedTxs: Promise<PopulatedTransaction>[] = [];
-    Object.keys(txChainsToInnerCalls).map((txChain: ChainName) => {
+  ): Promise<ethers.PopulatedTransaction[]> {
+    const txChainsToInnerCalls: Record<ChainName, CallData[]> = txs.reduce(
```

Comment:
Small nit: Strictly a matter of taste but I find using `reduce`'s generic param a bit nicer in some cases:
`const txChainsToInnerCalls = txs.reduce<Record<ChainName, CallData[]>>(`